yarn-project/aztec.js/src/account_manager/account_manager.ts

```diff
@@ -139,23 +139,27 @@ export class AccountManager {
   }
 
   /**
-   * Returns the pre-populated deployment method to deploy the account contract that backs this account.
+   * Returns the pre-populated ContractSetupMethods which can prepare a tx to
+   * initialize and/or publish this account's account contract (depending on the contract's design).
    * If no wallet is provided, it uses a signerless wallet with the multi call entrypoint
-   * @param deployWallet - Wallet used for deploying the account contract.
-   * @returns A DeployMethod instance that deploys this account contract
+   * @param deployWallet - Wallet used for any txs that are needed to
+   * set up the account contract for use.
+   * @returns A ContractSetupMethods instance that can set up this account contract for use
    */
   public async getDeployMethod(deployWallet?: Wallet): Promise<DeployMethod> {
     const artifact = await this.accountContract.getContractArtifact();
 
-    if (!(await this.isDeployable())) {
-      throw new Error(`Account contract ${artifact.name} does not require deployment.`);
+    if (!(await this.hasInitializer())) {
+      // TODO: there should be a path which enables an account contract's class & instance to be published,
+      // even if the account contract doesn't have an initializer function. This should not throw.
```

Comment:
If there is no issue created for this this will never get tackled so pls create an issue and link it here